### DISCUSSION TITLE
Show Field in "Grid View" but Hide in Detail View

### DISCUSSION THREAD

**Discussion by cancan101** [pimcore/pimcore#16926] [Q&A / Support] [answered]
I would like to show a field in the "Grid View", but hide that field in the "Detail View". Previously I had the field marked as ["Invisible"](https://github.com/pimcore/pimcore/blob/7812e7ddb5933a50f015b031419278f2fad3e5de/doc/05_Objects/01_Object_Classes/01_Data_Types/README.md?plain=1#L143) to hide the field in the Detail View. To add the field to the Grid View, I checked ["Visible in GridView"](https://github.com/pimcore/pimcore/blob/7812e7ddb5933a50f015b031419278f2fad3e5de/doc/05_Objects/01_Object_Classes/01_Data_Types/README.md?plain=1#L144-L145) but that does not show the field without also unchecking "Invisible". Once "Invisible" is unchecked, however, the field also shows in the Detail View (which I don't want). 

Is there any setting to show a Field in the Grid View without also showing in the Detail View?

**Comment by markus-moser** [comment, marked as answer]
This is not possible via the class editor. I see 2 reasonable options to solve this:

1. Create a custom object layout without the field which should not be visible on the data object detail and assign this via user permissions.

2. Hook into the pre-send-data event of the backend object detail json call and overrule the visible flag there.

Dynamically changing the json helped me a lot in the past for making things like this a little bit more flexible in my projects based on some logic but for sure you should not overuse this approach to keep things simple.
